extensions/ql-vscode/src/run-queries.ts

```diff
@@ -157,15 +157,22 @@ export class QueryInfo {
   }
 
   /**
-   * Holds if this query should produce interpreted results.
+   * Holds if this query can in principle produce interpreted results.
    */
-  async hasInterpretedResults(): Promise<boolean> {
+  async canHaveInterpretedResults(): Promise<boolean> {
     const hasMetadataFile = await this.dbItem.hasMetadataFile();
     if (!hasMetadataFile) {
       logger.log("Cannot produce interpreted results since the database does not have a .dbinfo or codeql-database.yml file.");
     }
     return hasMetadataFile;
   }
+
+  /**
+   * Holds if this query actually has produced interpreted results.
+   */
+  hasInterpretedResults(): boolean {
+    return fs.existsSync(this.resultsPaths.interpretedResultsPath);
```

Comment:
Can we make this function async? Otherwise, this is blocking.

Reply:
Ah, I was afraid the `getTreeItem` had to by synchronous, but it looks like vscode permits it to return a `Promise`. Happy to make it `async` then.